Tests/SentryTests/Helper/SentryDeviceTests.mm

```diff
@@ -0,0 +1,188 @@
+#import "SentryDevice.h"
+#import <XCTest/XCTest.h>
+
+#define SENTRY_ASSERT_EQUAL(actualString, expectedString)                                          \
+    XCTAssert([expectedString isEqualToString:actualString], @"Expected %@ but got %@",            \
```

Comment:
`XCTAssertEquals` only checks equivalence using `==`, which for `NSString`s only compares pointers. You must always compare string _contents_ using `-[isEqualToString:]`. `==` can appear to work if you're comparing interned string literals to each other since they'll be the same pointer if the string literals have the same contents, but will break if you swap out one of them for a dynamically generated string, like what we're comparing here (NSStrings constructed from C strings).

Pop this into a test case and run it to verify:
```objective-c
- (void)testStringLiterals {
    NSString *a = @"a";
    NSString *a2 = @"a";
    NSString *b = @"b";
    NSString *a3 = [NSString stringWithUTF8String:"a"];

    XCTAssertEqual(a, a2); // true; @"a" and @"a" are interned string constants, so the pointers a and a2 are equivalent
    XCTAssertEqual(a, b); // false; @"a" and @"b" are different interned string constants, so a and b are different pointers to different objects
    XCTAssertEqual(a, a3); // false; a (@"a") is an interned string constant, while a3 is a dynamically generated string, thus not the same object as the interned constant so pointer equality comparison fails
}
```